src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/codeSmell/MagicNumberInspector.java

```diff
@@ -0,0 +1,290 @@
+package com.kalessil.phpStorm.phpInspectionsEA.inspectors.codeSmell;
+
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.PsiReference;
+import com.intellij.psi.tree.IElementType;
+import com.jetbrains.php.lang.lexer.PhpTokenTypes;
+import com.jetbrains.php.lang.parser.PhpElementTypes;
+import com.jetbrains.php.lang.psi.elements.*;
+import com.jetbrains.php.lang.psi.elements.impl.PhpExpressionImpl;
+import com.kalessil.phpStorm.phpInspectionsEA.openApi.BasePhpElementVisitor;
+import com.kalessil.phpStorm.phpInspectionsEA.openApi.BasePhpInspection;
+import com.kalessil.phpStorm.phpInspectionsEA.options.OptionsComponent;
+import com.kalessil.phpStorm.phpInspectionsEA.utils.BinaryExpressionUtil;
+import com.kalessil.phpStorm.phpInspectionsEA.utils.ElementTypeUtil;
+
+import javax.swing.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class MagicNumberInspector extends BasePhpInspection {
+    private static final String message = "Magic number should be replaced by a constant.";
+
+    boolean optionCheckOnMultiplier = true;
+    boolean optionCheckOnProperties = true;
+    boolean optionCheckOnParameters = true;
+    boolean optionCheckOnArguments  = true;
+
+    @NotNull
+    public String getShortName() {
+        return "MagicNumberInspection";
+    }
+
+    @Override
+    @NotNull
+    public PsiElementVisitor buildVisitor(@NotNull final ProblemsHolder problemsHolder, final boolean isOnTheFly) {
+        return new BasePhpElementVisitor() {
+            @Override
+            public void visitPhpReturn(final PhpReturn returnStatement) {
+                final PsiElement returnArgument = returnStatement.getArgument();
+
+                if (isNumeric(returnArgument)) {
+                    registerProblem(returnArgument);
+                }
+            }
+
+            @Override
+            public void visitPhpBinaryExpression(final BinaryExpression expression) {
+                if (Objects.equals(expression.getOperationType(), PhpTokenTypes.opMUL)) {
```

Comment:
It was a IDE advice (I have enabled all and now I am filtering some haha). But it in this case we have no problem, it will just calls the `equals()` from operation type (that probabily uses directly the `==`). But I will change that.